lib/sparse/src/index/inverted_index/inverted_index_ram_builder.rs

```diff
@@ -46,11 +47,33 @@ impl InvertedIndexBuilder {
 
     /// Consumes the builder and returns an InvertedIndexRam
     pub fn build(self) -> InvertedIndexRam {
+        if self.posting_builders.is_empty() {
+            return InvertedIndexRam {
+                postings: vec![],
+                total_sparse_size: self.total_sparse_size,
+                vector_count: self.vector_count,
+            };
+        }
+
+        let num_posting_lists = self.posting_builders.len();
+        debug!(
+            "building inverted index with {} posting lists",
+            num_posting_lists
+        );
+        let mut new_vectors_count = 0;
+
         let mut postings = Vec::with_capacity(self.posting_builders.len());
         for posting_builder in self.posting_builders {
-            postings.push(posting_builder.build());
+            let posting_list = posting_builder.build();
+            new_vectors_count += posting_list.elements.len();
```

Comment:
the sum of posting length is not equal to the vector count

points have their dimension spread over several posting lists

Reply:
ohh right! makes sense. Each posting list represents a dimension and we already have `self.vector_count` which was incremented by `add()`